engine.io-server/src/main/java/io/socket/engineio/server/EngineIoSocket.java

```diff
@@ -303,17 +309,15 @@ private void flush() {
     }
 
     private void resetPingTimeout() {
```

Comment:
Would it make sense to make this a synchronized method?

Reply:
Having a look at the code again, it is theoretically possible for two threads to come through here. 
The effects of this could be that the current timer is cancelled, twice which is fine, and then both come through and start a new one, but one of them would not be assigned to the reference and therefor never cancelled. Thus the connection would drop.
It feels safe and robust to add synchronization to this method.